lib/mail_room/jwt.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'securerandom'
+
+module MailRoom
+  # Responsible for validating and generating JWT token
+  class JWT
+    DEFAULT_ISSUER = 'mailroom'
+    DEFAULT_ALGORITHM = 'HS256'
+
+    attr_reader :header, :secret_path, :issuer, :algorithm
+
+    def initialize(header:, secret_path:, issuer:, algorithm:)
+      @header = header
+      @secret_path = secret_path
+      @issuer = issuer || DEFAULT_ISSUER
+      @algorithm = algorithm || DEFAULT_ALGORITHM
+    end
+
+    def valid?
+      [@header, @secret_path, @issuer, @algorithm].none?(&:nil?)
+    end
+
+    def token
+      return nil unless valid?
+
+      secret = Base64.strict_decode64(File.read(@secret_path).chomp)
+      payload = { nonce: SecureRandom.hex(12), iss: @issuer }
```

Comment:
Can `iss` be `issuer`? If another user wanted to use JWT auth, their receiving service would have to follow this protocol?

Reply:
I think `iss` is a standard JWT claim: https://datatracker.ietf.org/doc/html/rfc7519#section-4.1

Reply:
Thanks @stanhu. Yes `iss` is a standard JWT claim. There are also other standard ones. Supporting all of them leading to a bunch of new redundant configurations at the moment. So, I think the one who need more can contribute in the future.